client-misk/src/main/kotlin/app/cash/backfila/client/misk/client/BackfilaClientLoggingSetup.kt

```diff
@@ -0,0 +1,62 @@
+package app.cash.backfila.client.misk.client
+
+import misk.logging.getLogger
+import org.slf4j.MDC
+
+interface BackfilaClientLoggingSetupProvider {
+  fun <T>withBackfillRunLogging(backfillName: String, backfillId: String, wrapped: () -> T) : T
+  fun <T>withBackfillPartitionLogging(backfillName: String, backfillId: String, partitionName: String, wrapped: () -> T) : T
+}
+
+class BackfilaClientNoLoggingSetupProvider : BackfilaClientLoggingSetupProvider {
+  override fun <T>withBackfillRunLogging(backfillName: String, backfillId: String, wrapped: () -> T) : T {
+    return wrapped.invoke()
+  }
+
+  override fun <T>withBackfillPartitionLogging(backfillName: String, backfillId: String, partitionName: String, wrapped: () -> T) : T {
+    return wrapped.invoke()
+  }
+}
+
+class BackfilaClientMDCLoggingSetupProvider : BackfilaClientLoggingSetupProvider {
+  companion object {
+    val log = getLogger<BackfilaClientMDCLoggingSetupProvider>()
+    const val MDC_BACKFILL_NAME = "backfill_name"
+    const val MDC_BACKFILL_ID = "backfill_id"
+    const val MDC_PARTITION_NAME = "partition_name"
+  }
+  override fun <T>withBackfillRunLogging(backfillName: String, backfillId: String, wrapped: () -> T) : T {
+    try {
+      MDC.put(MDC_BACKFILL_NAME, backfillName)
+      MDC.put(MDC_BACKFILL_ID, backfillId)
+    } catch (e: Exception) {
+      log.debug("Exception setting log context context", e)
```

Comment:
is this useful? also should e be the first arg here

Reply:
I think this is fine?
```
    /**
     * Log an exception (throwable) at the DEBUG level with an
     * accompanying message.
     *
     * @param msg the message accompanying the exception
     * @param t   the exception (throwable) to log
     */
    public void debug(String msg, Throwable t);
```